src/components/side_panel/side_panel/side_panel_store.ts

```diff
@@ -135,6 +136,41 @@ export class SidePanelStore extends SpreadsheetStore {
     this._openPanel("secondaryPanel", newPanelInfo, state);
   }
 
+  replace(
+    componentTag: string,
+    currentPanelKey: string,
+    initialPanelProps: SidePanelComponentProps = {}
+  ) {
+    const newPanelInfo = { initialPanelProps, componentTag, size: DEFAULT_SIDE_PANEL_SIZE };
+    const state = this.computeState(newPanelInfo);
+    if (!state.isOpen) {
+      return;
+    }
+    const expandMainPanelIfCollapsed = () => {
+      if (this.mainPanel?.isCollapsed) {
```

Comment:
why not expending it whatever happens?

Reply:
I think i's not that easy to follow what goes on in that panel :p 

Indeed it looks necessary :)